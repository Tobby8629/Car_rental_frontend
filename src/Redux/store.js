import { configureStore } from '@reduxjs/toolkit';
import UserSlice from './UserSlice';
import CarSlice from './CarSlice';
import ReservationSlice from './ReservationSlice';

const store = configureStore({
  reducer: {
    Users: UserSlice,
    Cars: CarSlice,
    Reserve: ReservationSlice,
  },
});

export default store;
