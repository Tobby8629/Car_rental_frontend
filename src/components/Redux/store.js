import { configureStore } from '@reduxjs/toolkit';
import UserSlice from './UserSlice';
import ReservationSlice from './ReservationSlice';

const store = configureStore({
  reducer: {
    Users: UserSlice,
    Reserve: ReservationSlice,
  },
});

export default store;
