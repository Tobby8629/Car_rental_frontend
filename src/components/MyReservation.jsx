import React, { useEffect, useState } from 'react';
import './componentsCss/delete.css';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { deleteReserve, getReserve } from '../Redux/ReservationSlice';

function DeleteCar() {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getReserve());
  }, [dispatch]);

  const cars = useSelector((state) => state.Reserve.reservations);

  const [buttondis, setbuttondis] = useState(false);
  const [prevdis, setprev] = useState(false);
  const [car, setcar] = useState(1);
  const perpage = 6;
  const last = car * perpage;
  const first = last - perpage;
  const each = cars.slice(first, last);
  const next = () => {
    if (car >= Math.round(cars.length / 6)) {
      setbuttondis(true);
    } else {
      setbuttondis(false);
      setcar(car + 1);
      setprev(false);
    }
  };

  const prev = () => {
    if (car <= 1) {
      setprev(true);
    } else {
      setprev(false);
      setcar(car - 1);
      setbuttondis(false);
    }
  };

  const Delete = async (id) => {
    try {
      await dispatch(deleteReserve(id));
      window.location.reload();
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <section className="delete myreserve">
      {each.length > 0 ? (
        <>
          <div className="header">
            <h3>Here is the List of Reservations</h3>
          </div>

          <div className="all">
            {each.map((e) => (
              <div className="each" key={e.id}>
                <div className="image">
                  <img src={e.image} alt={e.name} />
                </div>
                <div className="text">
                  <h4>{e.name}</h4>
                  <div className="price">
                    <div>
                      <h5>
                        city:
                        <span className="small">
                          {' '}
                          {e.city}
                        </span>
                      </h5>
                      <h5>
                        pickup:
                        <span className="small">
                          {' '}
                          {e.pickup}
                        </span>
                      </h5>
                    </div>
                    <div>
                      <h5>
                        date:
                        <span className="small">
                          {' '}
                          {e.pick_up}
                        </span>
                      </h5>
                      <h5>
                        return date:
                        <span className="small">
                          {' '}
                          {e.return_date}
                        </span>
                      </h5>
                    </div>
                  </div>
                  <button type="button" onClick={() => Delete(e.id)}>cancel reservation</button>
                </div>
              </div>
            ))}
          </div>
          <div className="btn">
            <button className="prev" type="button" disabled={prevdis} onClick={prev}>
              prev
            </button>
            <button className="next" type="button" disabled={buttondis} onClick={next}>
              next
            </button>
            <button className="next" type="button">
              <Link to="/reserve">Add Reservation</Link>
            </button>
          </div>
        </>
      ) : (
        <div>
          <h2>
            You do not have any reservation, please add a
            <Link to="/reserve">new Reservation</Link>
          </h2>
        </div>
      )}
    </section>

  );
}

export default DeleteCar;
