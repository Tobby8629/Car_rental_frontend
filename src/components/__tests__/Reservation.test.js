import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../../Redux/store';
import MyReservation from '../MyReservation';

test('MyReservation component matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <MyReservation />
      </BrowserRouter>
    </Provider>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
