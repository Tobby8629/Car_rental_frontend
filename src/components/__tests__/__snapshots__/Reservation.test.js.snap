// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyReservation component matches snapshot 1`] = `
<section
  className="delete myreserve"
>
  <div>
    <h2>
      You do not have any reservation, please add a
      <a
        href="/reserve"
        onClick={[Function]}
      >
         new Reservation
      </a>
    </h2>
  </div>
</section>
`;
